tools/topology/topology2/include/components/aec.conf

```diff
@@ -0,0 +1,160 @@
+#
+# Common widget aec
+#
+# A generic aec widget. All attributes defined herein are namespaced
+# by alsatplg to "Object.Widget.aec.N.attribute_name"
+#
+# Usage: this component can be used by instantiating it in the parent object. i.e.
+#
+# 	Object.Widget.aec."N" {
+# 		index		1
+# 		type		"effect"
+#	}
+#
+# Where N is the unique instance number for the aec object within the same alsaconf node.
+Define {
+	AEC_UUID	"a6:a0:80:b7:9f:26:6f:46:b4:77:23:df:a0:5a:f7:58"
+}
+
+Class.Widget."aec" {
+	#
+	# Pipeline ID for the aec widget object
+	#
+	DefineAttribute."index" {}
+
+	#
+	# aec object instance
+	#
+	DefineAttribute."instance" {}
+
+	#include common component definition
+	<include/components/widget-common.conf>
+
+	DefineAttribute."num_audio_formats" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	#
+	# aec widget UUID
+	#
+	DefineAttribute."uuid" {
+		type	"string"
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.uuid"
+	}
+
+	DefineAttribute."cpc" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	DefineAttribute."is_pages" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	# Attribute categories
+	attributes {
+		#
+		# The aec widget name would be constructed using the index and instance attributes.
+		# For ex: "aec.1.1" or "aec.10.2" etc.
+		#
+		!constructor [
+			"index"
+			"instance"
+		]
+
+		#
+		# mandatory attributes that must be provided when the class is instantiated
+		#
+		!mandatory [
+			"no_pm"
+			"uuid"
+			"process_type"
+		]
+
+		#
+		# immutable attributes cannot be modified in the object instance
+		#
+		!immutable [
+			"uuid"
+			"type"
+		]
+
+		#
+		# deprecated attributes should not be added in the object instance
+		#
+		!deprecated [
+			"preload_count"
+		]
+
+		#
+		# aec widget objects instantiated within the same alsaconf node must have unique
+		# instance attribute
+		#
+		unique	"instance"
+	}
+
+	num_audio_formats 2
+	num_input_audio_formats 4
+	num_output_audio_formats 2
+	Object.Base.input_audio_format {
+		1 {
+			pin_index               0
+		}
+		2 {
+			pin_index               1
+		}
+		3 {
+			pin_index               0
+			in_channels             4
+			out_channels            4
+			in_ch_cfg       $CHANNEL_CONFIG_QUATRO
+			in_ch_map       $CHANNEL_MAP_3_POINT_1
+			out_ch_cfg      $CHANNEL_CONFIG_QUATRO
+			out_ch_map      $CHANNEL_MAP_3_POINT_1
+		}
+		4 {
+			pin_index               1
+			in_channels             4
+			out_channels            4
+			in_ch_cfg       $CHANNEL_CONFIG_QUATRO
+			in_ch_map       $CHANNEL_MAP_3_POINT_1
+			out_ch_cfg      $CHANNEL_CONFIG_QUATRO
+			out_ch_map      $CHANNEL_MAP_3_POINT_1
```

Comment:
4 channels for echo cancellation? What does this mean? Are we talking about 4 channels for the reference, for the captured data, both, none?

Reply:
It is for future. If we don't have such requirement, I will remove it.